homeassistant/components/calendar/__init__.py

```diff
@@ -67,6 +67,23 @@
 VALID_FREQS = {"DAILY", "WEEKLY", "MONTHLY", "YEARLY"}
 
 
+def _has_timezone(*keys: Any) -> Callable[[dict[str, Any]], dict[str, Any]]:
+    """Assert that all datetime values have a timezone."""
+
+    def validate(obj: dict[str, Any]) -> dict[str, Any]:
+        """Validate that all datetime values have a timezone."""
+        for k in keys:
+            if (
+                (value := obj.get(k))
+                and isinstance(value, datetime.datetime)
+                and value.tzinfo is None
+            ):
+                raise vol.Invalid("Expected all values to have a timezone")
```

Comment:
Note to reviewer: This is not covered by tests because all the integrations are well conforming to not trigger this. If desired, we can update a test like google calendar to return incorrect data here.